src/asset-pipeline/functions/compressTextures.ts

```diff
@@ -0,0 +1,123 @@
+import { Document, Texture, Transform } from "@gltf-transform/core";
+import { TextureBasisu } from "@gltf-transform/extensions";
+import { listTextureSlots } from "@gltf-transform/functions";
+
+import { createDeferred, Deferred } from "../../engine/utils/Deferred";
+import TextureCompressionWorker from "./TextureCompressionWorker?worker";
+
+export function compressTextures(): Transform {
+  return async (doc: Document) => {
+    const root = doc.getRoot();
+
+    let nextJobId = 0;
+    const jobs: Map<number, Deferred<{ jobId: number; data: Uint8Array; mimeType: string }>> = new Map();
```

Comment:
might be nice to eventually create a generic job system for more stuff like this in the future :D